### PR TITLE
[OAP-1734][oap-native-sql]use non-codegen for sort with single key

### PR DESCRIPTION
**Pull request by rui-mo**
1. changes sort with one key inside several cols into non-codegen version. 
2. fixes data type issue and null value issue for InplaceSort.

Fixes: https://github.com/Intel-bigdata/OAP/issues/1734

### CODE REVIEW

**Comment by rui-mo** [comment]
@zhztheplayer Here is an initial version, and it only works for asc and null first case. The gaps are:
1. In function FinishInternal, need to support different sort cases.
2. Need to support string type.
3. move Appender class to a new file.

File: oap-native-sql/cpp/src/codegen/arrow_compute/ext/sort_kernel.cc [hunk 1]
```diff
@@ -630,18 +630,20 @@ class SortInplaceKernel : public SortArraysToIndicesKernel::Impl {
...
       auto length = (total_length_ - offset_) > batch_size_ ? batch_size_
                                                             : (total_length_ - offset_);
-      uint64_t count = 0;
+      uint64_t valid_count = 0;
+      uint64_t total_count = 0;
       if (offset_ >= nulls_total_) {
-        while (count < length) {
-          RETURN_NOT_OK(builder_0_->Append(result_arr_->GetView(offset_ + count++)));
+        while (total_count < length) {
+          RETURN_NOT_OK(builder_0_->Append(result_arr_->GetView(offset_ + total_count++)));
```
Comment (zhouyuan) [inline oap-native-sql/cpp/src/codegen/arrow_compute/ext/sort_kernel.cc:637]: the logic seems complicated... can you move the total_count++ to a separate line? so it should be much easier to review

File: oap-native-sql/cpp/src/codegen/arrow_compute/ext/sort_kernel.cc [hunk 2]
```diff
@@ -630,19 +631,36 @@ class SortInplaceKernel : public SortArraysToIndicesKernel::Impl {
     arrow::Status Next(std::shared_ptr<arrow::RecordBatch>* out) {
       auto length = (total_length_ - offset_) > batch_size_ ? batch_size_
                                                             : (total_length_ - offset_);
+      /* Here we take value from the sorted result_arr_ and append to builder.
+      valid_count is used to count the valid value, for accessing the valid value
+      in result_arr_.
+      total_count is used to count both valid and null value, for determing if all values
+      are appended to builder in while loop.
+      */
```
Comment (zhouyuan) [inline oap-native-sql/cpp/src/codegen/arrow_compute/ext/sort_kernel.cc:639]: great notes!

**Comment by rui-mo** [comment]
@zhouyuan @xuechendi verified on TPC-H. TPC-DS queries are all runnable now but having result mismatch, see https://github.com/Intel-bigdata/OAP/issues/1751.

**Comment by zhouyuan** [comment]
notes: 
- sort by multiple keys is using codegen code path
- sort by single key but w/o payload is using inplace sort
- sort by single key w/ payload is using non-codegen code path

The multiple keys sorting w/ different orders, w/ nulls first/last will be fixed in another PR

**Review by zhouyuan — APPROVED** [review]

**Pull request merged**